src/alfalfa/AlfalfaComponent.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef ALFALFA_COMPONENT_HPP
+#define ALFALFA_COMPONENT_HPP
+
+#include <json/json.h>
+#include <type_traits>
+
+#include "AlfalfaAPI.hpp"
+#include "ComponentBase.hpp"
+
+namespace openstudio {
+namespace alfalfa {
+  class ALFALFA_API AlfalfaComponent
+  {
+   public:
+    template <typename T, std::enable_if_t<std::is_base_of<ComponentBase, T>::value, bool> = true>
+    AlfalfaComponent(T component) : m_component(std::make_unique<T>(std::move(component))) {}
```

Comment:
seeing this while looking diagonally at this PR changes since I was out of office. Makes me wonder if I should investigate what's up or not. @kbenne I'm assuming you reviewed the changes already while I wasn't around and I don't need to?

Reply:
(I'm failing to see directly why this AlfalfaComponent class is needed. Instead of having ComponentBase (renamed to AlfalfaComponent probably, but basically an abstract class with the pure virtual methods) and just be done with it)

Reply:
Ahh, the biggest issue appears to be that you can't `boost::optional` an abstract base class. I can obviously use a `unique_ptr` but I don't think that translates to swig very well...

Reply:
@jmarrec It wouldn't hurt to have your touch on this for one more round. If you and @TShapinsky agree to a simplification then I'm all for that.